Meta/CMake/libweb_generators.cmake

```diff
@@ -111,14 +111,30 @@ function (generate_js_bindings target)
             list(GET BINDINGS_TYPES ${iter} bindings_type)
             get_property(include_paths DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
             list(TRANSFORM include_paths PREPEND -i)
+
+            # Ninja expects the target name in depfiles to be relative to CMAKE_BINARY_DIR, but ${bindings_src} is
+            # relative to CMAKE_CURRENT_BINARY_DIR. CMake >= 3.20 can do the rewriting transparently (CMP0116).
+            if(CMAKE_GENERATOR MATCHES "^Ninja" AND NOT POLICY CMP0116)
+                # FIXME: Drop this branch for cmake_minimum_required(3.20)
+                get_filename_component(full_path ${bindings_src} ABSOLUTE BASE_DIR ${CMAKE_CURRENT_BINARY_DIR})
+                file(RELATIVE_PATH depfile_target ${CMAKE_BINARY_DIR} ${full_path})
+            else()
+                if (POLICY CMP0116)
+                    cmake_policy(SET CMP0116 NEW)
```

Comment:
We should set CMake policies very early in the build. Before creating any targets even. Just need to make sure it's set for both serenity and Lagom.